src/commands/command.ts

```diff
@@ -16,17 +18,19 @@ export abstract class Command {
     abstract triggerCriteria: TriggerCriteria[];
     requiredPerms: PermissionsLevel = PermissionsLevel.USER;
 
-    abstract validate(message: Message, trigger: ConcreteTrigger, args: string[]): boolean;
-    abstract execute(message: Message, trigger: ConcreteTrigger, args: string[]): void;
+    abstract validate(message: Message, trigger: ConcreteTrigger): boolean;
+    abstract execute(message: Message, trigger: ConcreteTrigger): void;
 
-    checkTriggers(message: Message, keyword: string, args: string[]): ConcreteTrigger | null {
+    checkTriggers(message: Message): ConcreteTrigger | null {
         const trigger: Partial<ConcreteTrigger> = {
             command: this
         };
+        
+        trigger.args = message.content.split(/ +/g);
 
         for (const criteria of this.triggerCriteria) {
             if (typeof criteria === 'function') {
-                const res: string | null = criteria(message, keyword, args);
+                const res: string | null = criteria(message, trigger.args.shift()?.toLowerCase() || "", trigger.args);
```

Comment:
I think worth extracting to a variable for clarity, and then maybe leave the keyword as `args[0]` as well, only slicing off the prefix.